stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/core.py

```diff
@@ -337,25 +339,25 @@ async def post_search(
             self.item_serializer.db_to_stac(item, base_url=base_url) for item in items
         ]
 
-        # if self.extension_is_enabled("FieldsExtension"):
-        #     if search_request.query is not None:
-        #         query_include: Set[str] = set(
-        #             [
-        #                 k if k in Settings.get().indexed_fields else f"properties.{k}"
-        #                 for k in search_request.query.keys()
-        #             ]
-        #         )
-        #         if not search_request.fields.include:
-        #             search_request.fields.include = query_include
-        #         else:
-        #             search_request.fields.include.union(query_include)
-
-        #     filter_kwargs = search_request.fields.filter_fields
-
-        #     response_features = [
-        #         json.loads(stac_pydantic.Item(**feat).json(**filter_kwargs))
-        #         for feat in response_features
-        #     ]
+        if self.extension_is_enabled("FieldsExtension"):
+            if search_request.query is not None:
+                query_include: Set[str] = set(
+                    [
+                        k if k in Settings.get().indexed_fields else f"properties.{k}"
+                        for k in search_request.query.keys()
```

Comment:
One thing to be aware of here if you plan on eventually supporting Python 3.11 -- Python 3.11 changes the default `__str__` implementation for enums, so you _may_ have to define it yourself. See https://github.com/stac-utils/stac-fastapi/commit/c32d5bc4979961493eea4276548012f1155f8e86#diff-7e6336bc2ad3ff7b065d2980d4d70fb1e5292b8748be4b8ce65d1c61425a7ab8R69-R77 for an example.

Reply:
Good to know. Maybe now is the time to update